psh/date/date.c

```diff
@@ -0,0 +1,221 @@
+/*
+ * Phoenix-RTOS
+ *
+ * date - print or set the system date and time
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <sys/time.h>
+#include <unistd.h>
+
+#include "../psh.h"
+
+
+enum mode { skip,
+	help,
+	print,
+	set,
+	parse };
+
+
+static void psh_date_info(void)
+{
+	printf("print/set the system date and time");
+}
+
+
+static void psh_date_help(const char *prog)
+{
+	printf("Usage: %s [-h] [-s EPOCH] [-d @EPOCH] [+FORMAT]\n", prog);
+	printf("  -h:  shows this help message\n");
+	printf("  -s:  set system time described by EPOCH (POSIX time format)\n");
+	printf("  -d:  display time described by EPOCH (POSIX time format)\n");
+	printf("  FORMAT: string with POSIX date formatting characters\n");
+	printf("NOTE: FORMAT string not supported by options: '-s', '-d'\n");
+}
```

Comment:
Just one remark, for future. On small targets (and when running from `RAM`) we shall use version with limited help in size and common error strings as much as possible (the compiler may reuse common strings placed in `.rodata`), current approach gets `psh` growing.

For now, better approach is to use just one `printf` with multi-line string, that is one call to `printf` instead of six, and around 56 bytes saved :) verified for `imxrt117x`.

```suggestion
	printf(
		"Usage: %s [-h] [-s EPOCH] [-d @EPOCH] [+FORMAT]\n"
		"  -h:  shows this help message\n"
		"  -s:  set system time described by EPOCH (POSIX time format)\n"
		"  -d:  display time described by EPOCH (POSIX time format)\n"
		"  FORMAT: string with POSIX date formatting characters\n"
		"NOTE: FORMAT string not supported by options: '-s', '-d'\n",
		prog
	  );
```

Reply:
Your suggestion was implemented. Thanks for that remark!